### PR TITLE
Pull hook

### PR DESCRIPTION
**Pull request by minhducsun2002**
Rewrite of #3556.

### CODE REVIEW

**Comment by ragesoss** [comment]
I tested this out and it works nicely. Have a look before I merge it, @bwreid ?

**Comment by bwreid** [comment]
Unfortunately `realpath` doesn't work for me on Mac.  :\  Can something similar be done with `pwd`? For example:
```
SCRIPT_ABSOLUTE_PATH=$(pwd -P $SCRIPT_PATH)
```

**Comment by minhducsun2002** [comment]
`readlink` should be portable.

**Comment by ragesoss** [comment]
I haven't had a chance to check it on Mac yet, but it already worked for Linux and this seems like it ought to work for Mac as well. I'll verify next week before merging it.

**Comment by minhducsun2002** [comment]
Well I checked with a Hackintosh user and looks like `readlink` is available. It should work.

**Comment by bwreid** [comment]
I'm sorry, I should've gotten to this on Friday.

`readlink` still doesn't perform correctly for me. The command is available but I get the following output:

```
$ ./.git-hooks/install-pull-update-deps
readlink: illegal option -- m
usage: readlink [-n] [file ...]
This will overwrite your current post-merge hook with /hooks/pull-update-deps.
cp: /../.git/hooks/post-merge: No such file or directory
```

File: .git-hooks/install-pull-update-deps [hunk 1]
```diff
@@ -1,8 +1,37 @@
...
+            link=$t
+        fi
+    done
+    realpath="$PWD/$(basename "$link")"
+    cd "$ourpwd"
+    echo "$realpath"
+}
+
+# ===========
+
```
Comment (ragesoss) [inline .git-hooks/install-pull-update-deps]: This seems like a lot of code to keep in the repo for such a narrow use case. I think a better approach would be to just document how to install the git hook manually by copying it to the correct location, without including a script to install it.

**Comment by minhducsun2002** [comment]
Done.

**Comment by ragesoss** [comment]
This cool with you @bwreid ?

**Comment by bwreid** [comment]
Yep, this works. Thanks for the changes!

**Pull request merged**